Regex improvements break Java 6 compatibility

This [commit](https://github.com/querydsl/querydsl/commit/f5502ef6fed62b3666710b26c68d64640f669cea) breaks compatibility with Java 6 runtimes trying to load the `TemplateFactory`:

```
Caused by: java.util.regex.PatternSyntaxException: Unknown look-behind group near index 15
\{(?<premod>%?%?)(?<index>\d+)(?<postmod>[slu%]?%?)\}
               ^
    at java.util.regex.Pattern.error(Pattern.java:1713)
    at java.util.regex.Pattern.group0(Pattern.java:2505)
    at java.util.regex.Pattern.sequence(Pattern.java:1806)
    at java.util.regex.Pattern.expr(Pattern.java:1752)
    at java.util.regex.Pattern.compile(Pattern.java:1460)
    at java.util.regex.Pattern.<init>(Pattern.java:1133)
    at java.util.regex.Pattern.compile(Pattern.java:823)
    at com.mysema.query.types.TemplateFactory.<clinit>(TemplateFactory.java:38)
    ... 64 more
```

This happens on a JDK 6 on Mac:

```
Serendipity:spring-data-jpa @ master olivergierke $ java -version
java version "1.6.0_65"
Java(TM) SE Runtime Environment (build 1.6.0_65-b14-462)
Java HotSpot(TM) 64-Bit Server VM (build 20.65-b04-462, mixed mode)
```

as well as on Linux boxes (see this [build log](https://build.spring.io/download/SPRINGDATAJPA-DATAJPA-JOB1/build_logs/SPRINGDATAJPA-DATAJPA-JOB1-688.log) for example). Unfortunately the commit doesn't contain any explanation what the context of this improvement is or what the drivers behind it were.

--- a/querydsl-core/src/main/java/com/mysema/query/types/TemplateFactory.java
+++ b/querydsl-core/src/main/java/com/mysema/query/types/TemplateFactory.java
@@ -36,9 +36,9 @@ public class TemplateFactory {
     private static final Constant<String> PERCENT = ConstantImpl.create("%");
 
     private static final Pattern elementPattern = Pattern.compile("\\{"
-            + "(?<premod>%?%?)"
-            + "(?<index>\\d+)"
-            + "(?<postmod>[slu%]?%?)"
+            + "(%?%?)"
+            + "(\\d+)"
+            + "([slu%]?%?)"
             + "\\}");
 
     private final Map<String,Template> cache = new ConcurrentHashMap<String,Template>();
@@ -177,9 +177,9 @@ public class TemplateFactory {
                 if (m.start() > end) {
                     elements.add(new Template.StaticText(template.substring(end, m.start())));
                 }
-                String premodifiers = m.group("premod").toLowerCase(Locale.ENGLISH);
-                int index = Integer.parseInt(m.group("index"));
-                String postmodifiers = m.group("postmod").toLowerCase(Locale.ENGLISH);
+                String premodifiers = m.group(1).toLowerCase(Locale.ENGLISH);
+                int index = Integer.parseInt(m.group(2));
+                String postmodifiers = m.group(3).toLowerCase(Locale.ENGLISH);
                 boolean asString = false;
                 Function<Object, Object> transformer = null;
                 switch (premodifiers.length()) {

